Client/Frontend/Browser/DefaultBrowserIntroManager.swift

```diff
@@ -0,0 +1,60 @@
+// Copyright 2020 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import Shared
+import BraveShared
+
+private typealias IntroPrefs = Preferences.DefaultBrowserIntro
+
+struct DefaultBrowserIntroManager {
+    /// This function should be called when the app is initialized.
+    /// It determines whether we should show default browser intro popup or not and sets corresponding preferences accordingly.
+    /// Returns true if the popup should be shown.
+    @discardableResult
+    static func prepareAndShowIfNeeded(isNewUser: Bool, launchDate: Date = Date()) -> Bool {
```

Comment:
Why do we need the launchDate? We are calling this static function only at didFinishLaunching right ?

Reply:
It is for tests only, to test if launching it at future date will behave correctly